front_end/handlers/BaseUserHandler.py

```diff
@@ -0,0 +1,58 @@
+from helper import *
+from tornado.web import *
+import traceback
+from content import *
+import contextvars
+
+
+settings_dict = load_yaml_dict(read_file("/Settings.yaml"))
```

Comment:
Let's not have any code that is outside of each handler class, other than import statements.

Put these lines of code within the class and uses these objects as properties/fields of the class. Then classes that inherit from this class can just retrieve settings_dict or content or whatever from these properties/fields rather than having to include code for declaring these objects in each inheriting class.

Please let me know if that doesn't make sense.

Reply:
It's possible that I'm misunderstanding your instructions because none of the functions inside HomeHandler can see these variables when I declare them like so. Should I be using ```self.settings_dict = ...``` for each variable?
<img width="720" alt="image" src="https://user-images.githubusercontent.com/61362908/128541191-87a9c112-2aae-4f41-9e93-c040998927ee.png">
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/61362908/128541377-ad707492-3373-4bb7-85f4-fa90031bd0c9.png">

Reply:
I'm pretty sure you will need to define a [constructor](https://www.geeksforgeeks.org/constructors-in-python/) for the HomeHandler class and place those four lines of code inside the constructor. Then when you want to access those variables from other functions in the class, you would use `self.settings_dict` or whatever.

Reply:
With this approach, any inheriting classes should also be able to access the variables using `self.` syntax.

Reply:
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/61362908/128550924-42fdec44-7a02-43e8-b6a8-ae5176d30f86.png">
<img width="752" alt="image" src="https://user-images.githubusercontent.com/61362908/128550955-b96748f6-3734-40aa-8811-c31785c47668.png">
It hasn't been too happy about constructors so far, this error log is all I have to go off and I can't think of a place where my constructor has 2 or more arguments (so far I have written each as ```def __init__(self):```). Without delving into this too much can you think of a reason why I'm getting these errors?

Reply:
Interesting. It must be related to the way the Tornado web server instantiates those classes. So maybe we can't do it this way. There is a prepare function, so perhaps you could do it there?

Reply:
Looks like that was the ticket! Thank you!